### ISSUE TITLE
HTMLView Android

### ISSUE THREAD

**Issue by b4rtt** [NativeScript/NativeScript#5367]
Hello, I am creating app for iOS and Android together and I found bug on my Android emulator.
Android can't render HTML tags. iOS is working great.

![snimek obrazovky 2018-02-01 v 11 51 57](https://user-images.githubusercontent.com/4247839/35674937-95fca7fc-0746-11e8-8d35-b2c4db2e0f67.png)

![img_0802](https://user-images.githubusercontent.com/4247839/35675096-2270b106-0747-11e8-886b-4fec820bd9d6.PNG)

**Comment by NickIliev** [comment]
@b4rtt I am guessing that you are trying to create an HTML `table`.
The thing is that HTML View is not supporting all HTML tags as in its native implementation lays `android.widget.TextView` (for Android)

The solution would be to use [WebView](https://docs.nativescript.org/cookbook/ui/web-view) and pass the string with your HTML table.

For example, this would render as expected on both iOS and Android.
```
<WebView height="400" src="{{ htmlString }}" />

// where htmlString is bind to
this.htmlString = "<table style='width:100%; font-size:26px'><tr><th>Firstname</th><th>Lastname</th> <th>Age</th></tr><tr><td>Jill</td><td>Smith</td> <td>50</td></tr><tr><td>Eve</td><td>Jackson</td><td>94</td></tr></table>";
```

Playground demonstration of the above can be found [here](https://play.nativescript.org/?template=play-tsc&id=pVWZaD)

**Comment by b4rtt** [comment]
@NickIliev Yes thats table. I know about WebView and that is better to use it. But main reason why I am using HTMLView is auto height depending of height of HTML content. My content (tables f.e) could have different count of rows and scrolling kills whole my UX. :(

**Comment by ManuelSch** [comment]
So is there any way in displaying a table (only containing data, no functionality) with native elements? I've read the [thread on discourse ](https://discourse.nativescript.org/t/gridlayout-for-table-very-laggy/4226/14) but the answers were pretty disappointing.

Grid Layouts seem to be have a very poor performance for big tables and a solution with List Views does not ensure that the column widths are equal for each row. And I don't like using a HTMLView as it defeats the reason why I chose NativeScript in the first place.

**Comment by NickIliev** [comment]
@ManuelSch I would use GridLayout without concerns and if the data is bigger I would combine it with ListView or RadListView and let the item templates take care of the performance (as ListView support recycling). 

What do you mean by " big tables"? You should keep in mind that you are creating a mobile application that will run on a handheld device so what could be the largest table you would want to visualize at any moment without using any kind of template structure (e.g. ListView).  Performance for the mobile application is crucial as they will run not on powerful PCs but still, most performance issues are related to not following mobile concepts. See this comment [here ](https://discourse.nativescript.org/t/gridlayout-for-table-very-laggy/4226/14?u=nickiliev)which describes the issue very well.

The HTML view is simply a base string that can be converted to base HTML. For advanced HTML usage, I would choose and use WEebView. Still, the better solution is not to use WebView when possible as you the main advantage of NativeScript is that you can use the native mobile widgets and concepts.

**Comment by ManuelSch** [comment]
@NickIliev In my case I have 10-20 rows and 5 columns. On my test device (Samsung J1) it takes 2-3 seconds just to render the table, which seems too long to me even for a low-end phone from 2016.

How do you suggest combining it with a ListView for this purpose? If I split the Grid Layout into multiple recyclable parts the column widths won't be the same, as the length of the cell content may vary.

**Comment by NickIliev** [comment]
@ManuelSch can you provide a sample Playground demonstrating your case with some dummy data?

**Comment by CatchABus** [comment]
Closing as not planned since the supported tags depend on each platform.
